### ISSUE TITLE
Connection should have references to the handles

### ISSUE THREAD

**Issue by nadenf** [wbkd/react-flow#589]
**Issue:**
For nodes that have multiple handles it is not possible to know to which handle was connected during an onConnect event.

**Expected:**
The signature of Connection should have a reference to the source and target handles:
```
interface Connection {
  source: ElementId | null;
  sourceHandle: HandleId | null;
  target: ElementId | null;
  targetHandle: HandleId | null;
}
```

**Actual:**
The signature of Connection assumes all nodes have a single handle e.g.

```
interface Connection {
  source: ElementId | null;
  target: ElementId | null;
}
```

**Comment by chrtze** [comment]
This will be fixed in the next version. You can already use it with `npm install react-flow-renderer@next`. #657

**Comment by chrtze** [comment]
released in [7.0.0](https://github.com/wbkd/react-flow/releases/tag/7.0.0)
